docs/development/releasing.rst

```diff
@@ -1,233 +1,386 @@
+the IERS tables may result in test failures, this update should be done via a
+pull request to the ``main`` branch, and then backported to the release branch.
 
-   At this point, you will likely need to update the Astropy ``.mailmap`` file,
-   which maps contributor emails to names, as there are often contributors who
-   are not careful about using the same e-mail address for every commit, meaning
-   that they appear multiple times in the contributor list above, sometimes with
-   different spelling, and sometimes you may also just see their GitHub username
-   with no full name.
+.. _release-procedure-update-whatsnew:
 
-   The easiest way to get a full list of contributors and email addresses is
-   to do::
+Updating the What's new and contributors
+----------------------------------------
 
-      git shortlog -n -s -e
+Make sure to update the "What's new"
+section with the stats on the number of issues, PRs, and contributors.
+Since the What's New for the major release is now only present in the release
+branch, you should switch to it to, e.g.::
 
-   Edit the ``.mailmap`` file to add entries for new email addresses for already
-   known contributors (matched to the appropriate canonical name/email address).
-   You can also try and investigate users with no name to see if you can determine
-   their full name from other sources - if you do, add a new entry for them in
-   the ``.mailmap`` file. Once you have done this, you can re-run the
-   ``generate_releaserst.xsh`` script (you will likely need to iterate a few times).
-   Once you are happy with the output, copy it into the 'What's new' page for
-   the current release and commit this. E.g., ::
+   $ git checkout v5.0.x
 
-      $ git add docs/whatsnew/5.0.rst
-      $ git commit -m "Added contributor statistics and names"
+To find the statistics and contributors, use the `generate_releaserst.xsh`_
+script. This requires `xonsh <https://xon.sh/>`_ and `docopt
+<http://docopt.org/>`_ which you can install with::
 
-   Update the ``docs/credits.rst`` file to include any new contributors from the
-   above step, and commit this and the ``.mailmap`` changes::
+   pip install xonsh docopt
 
-      $ git add .mailmap
-      $ git add docs/credits.rst
-      $ git commit -m "Updated list of contributors and .mailmap file"
+You should then run the script in the root of the astropy repository as follows::
 
-   This last commit should be forward-ported to the ``main`` branch.
+   xonsh generate_releaserst.xsh 4.3 v5.0.dev \
+                                 --project-name=astropy \
+                                 --pretty-project-name=astropy \
+                                 --pat=<a GitHub personal access token>
 
-#. Push the release branch back to GitHub, e.g.::
+The first argument should be the last major version (before any bug fix
+releases, while the second argument should be the ``.dev`` tag that was just
```

Comment:
this would miss anyone contributing e.g a new bugfix that ends up in RC after branching, wouldn't it? Usually there are no newcomers in that category, but there could be.

Reply:
Maybe that can be fixed by changing:

```
xonsh generate_releaserst.xsh 4.3 v5.0.dev
```

to be something like

```
xonsh generate_releaserst.xsh 4.3 v5.0.x
```

to include all the latest commits on v5.0.x? I'm not sure, maybe @Cadair can give some advice here?

Reply:
Surely those bug fixes on the RC would also have been merged to the main branch? The reason you don't want to use the tag is that the tag isn't on the same branch as the rest of the commits you are measuring.

Reply:
I think the issue is then people who did those fixes wouldn't get included in the credits page until the *next* release? I'll open a separate issue about this so that we don't hold back this PR.

Reply:
See https://github.com/astropy/astropy/issues/12866